### ISSUE TITLE
What is the connection between GFLOPs value and GPU RAM value for Mask RCNN?

### ISSUE THREAD

**Issue by zerok01** [open-mmlab/mmdetection#4472]
Hi, when I run `python tools/get_flops.py configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py --shape 800` command, I get above result.

```
==============================
Input shape: (3, 800, 800)
Flops: 181.09 GFLOPs
Params: 44.17 M
==============================
```
When I change shape size, Params value does not change but Floops value increases. But  when I set shape value is 2000, I get Out of Memory(GPU RAM 8 GB).

In my mind, I could not establish the connection between the Flops value and GPU RAM. So what is the Ram equivalent of Flops value according to Mask_rcnn_r50_fpn_1x_coco config? Do I need to do RAM / Flops directly? Is there anyone who can explain this to me? Thank you(Sorry for my bad English)

**Comment by hhaAndroid** [comment]
@zerok01 flops are related to image resolution,  model forward will be called when you are run tools/get_flops.py

**Comment by zerok01** [comment]
> @zerok01 flops are related to image resolution, model forward will be called when you are run tools/get_flops.py

Thank you for your answer. But what I wanted to ask is KiloByte or MegaByte equivalent of the Flops value. So for example, can we say 1 GLOPS equals 1 GigaByte? Is there a formula for this? Actually, this is what I want to learn.

**Comment by hhaAndroid** [comment]
@zerok01  FLOPs is Floating-point Operations, MFLOPs is Mega FLOPs, GFLOPs is Giga FLOPs.  
The formula you can reference  [Pruning Convolutional Neural Networks for Resource Efficient Inference](https://arxiv.org/pdf/1611.06440v2.pdf) 
The  code you can reference [flops_to_string](https://github.com/open-mmlab/mmcv/blob/master/mmcv/cnn/utils/flops_counter.py) flops_to_string function

**Comment by ZwwWayne** [comment]
The connection between FLOPs and GPU memory cost is not very clear actually. It depends on the implementation of the model and PyTorch libraries. Usually, under similar conditions (model input etc.), models with small FLOPs should have small GPU memory cost.
